docs/components/concepts/job-workers.md

```diff
@@ -125,6 +121,30 @@ To help visualize the process in general, here is a sequence diagram which shows
+
+To avoid your workers being overloaded with too many jobs, e.g. running out of memory, the workers rely on the [built-in gRPC flow control mechanism](https://grpc.io/docs/guides/flow-control/) (or, if lacking for your language of choice, [the built-in HTTP/2 stream flow control](https://httpwg.org/specs/rfc7540.html#FlowControl), e.g. Golang implementatio of gRPC).
+
+Essentially, as jobs are pushed downstream from the broker to the client, they're first buffered in the gateway where the direct client connection reside. The gateway only sends as much data as the client can consume over a specific connection. If it notices its send buffers fill up, it will mark a client as `not-ready`.
+
+If a client is not ready to receive a job, the gateway will instead try to re-route the job to another, logically equivalent worker. If this fails (e.g. all workers connected to a specific gateway are not ready), the job is returned to the broker. There, it may be retried to another gateway, if and only if it has a logically equivalent worker.
+
+#### Detecting back pressure
+
+There are different ways to detect back pressure.
+
+On the client side, you can use the job worker metrics to do so. For example, by subtracting the rate of handled jobs (i.e. `zeebe.client.worker.job.handled`) from the rate of activated jobs (i.e. `zeebe.client.worker.job.activated`), you can estimate the rate of queued jobs. If this is too close to the `maxJobsActive` consistently, this may indicate you need to scale your worker deployment.
+
+On the server side (e.g. if you're running a self-managed cluster), you can measure the rate of jobs which are not pushed due to clients which are not ready via the metric `zeebe_broker_jobs_push_fail_try_count_total{code="BLOCKED"}`. If the rate of this metric is high for a sustained amount of time, it may be a good indicator that you need to scale your workers. Unfortunately, on the server side we don't differentiate between clients, so this metric doesn't tell you which worker deployment needs to be scaled. We thus recommend using client metrics whenever possible.
+
+#### Implementing backpressure
+
+If you're using the raw `StreamActivatedJobs` RPC, or want to add support for this to your client of choice, the criteria to apply back pressure is to stall the underlying HTTP/2 transport. To do so, you may need to block the thread in which the gRPC stream is running (e.g. Java), or suspend the coroutine (e.g. Kotlin, Go). Once the transport stops receiving, this will cause the gateway's send buffers to fill up, and effectively apply back pressure.
+
+If you wish to test this, you can do so by simulating a very slow worker with your new implementation. Then start generating many jobs on the server side (e.g. create many process instances with lots of jobs). You should then observe back pressure via server side metrics, or many `Job.YIELD` commands being written to the log.
+
+Refer to the Java and Go implementations for more on this.
```

Comment:
❓ Do you want to link here maybe?

Reply:
Hm, where exactly? I could link to the repo, but unfortunately I don't know how to grab the right version (e.g. if the user is seeing 8.3, this should point to 8.3). Or we link to the docs for each? But there we don't really talk about the implementation :thinking: 

I can just link to the general Zeebe repo I guess, but again it'll be unversioned :shrug: